ethcore/src/client/client.rs

```diff
@@ -451,24 +458,24 @@ impl Client {
 		};
 
 		// Enact Verified Block
-		let last_hashes = self.build_last_hashes(header.parent_hash().clone());
-		let db = self.state_db.lock().boxed_clone_canon(header.parent_hash());
+		let last_hashes = client.build_last_hashes(header.parent_hash().clone());
+		let db = client.state_db.lock().boxed_clone_canon(header.parent_hash());
 
 		let is_epoch_begin = chain.epoch_transition(parent.number(), *header.parent_hash()).is_some();
 		let enact_result = enact_verified(block,
 			engine,
-			self.tracedb.read().tracing_enabled(),
+			client.tracedb.read().tracing_enabled(),
 			db,
 			&parent,
 			last_hashes,
-			self.factories.clone(),
+			client.factories.clone(),
 			is_epoch_begin,
 		);
 		let mut locked_block = enact_result.map_err(|e| {
 			warn!(target: "client", "Block import failed for #{} ({})\nError: {:?}", header.number(), header.hash(), e);
 		})?;
 
-		if header.number() < self.engine().params().validate_receipts_transition && header.receipts_root() != locked_block.block().header().receipts_root() {
+		if header.number() < engine.params().validate_receipts_transition && header.receipts_root() != locked_block.block().header().receipts_root() {
 			locked_block = locked_block.strip_receipts();
```

Comment:
this is interesting. 

`import_verified_blocks` calls this function... and this function replaces receipt_root in block_header? Can it possibly lead to invalid block being imported?